jwst/residual_fringe/utils.py

```diff
@@ -1,6 +1,7 @@
 import numpy as np
 import math
 import numpy.polynomial.polynomial as poly
+import matplotlib.pyplot as plt
```

Comment:
This adds matplotlib as a dependency. 
How important is it to have it in operations? 
If it's meant to be used by an end  user only function, it should be added as an optional dependency and wrapped  in an `if ...else` block. Also the plotting functionality in the functions should check if matplotlib is available and be skipped with a warning if not.

Reply:
Removed dependency.